### PR TITLE
Add API Key support for all JWT points

### PR DESCRIPTION
**Pull request by Pasarus**
Closes None, part of https://github.com/fiaisis/run-detection/pull/343

## Description
Adjusts the JWT API authentication to allow an API key to be given instead of a JWT. Adds support for the API being treated as user_number 1 with staff level permissions for all authentication purposes.

### CODE REVIEW

File: fia_api/core/auth/tokens.py [hunk 1]
```diff
@@ -80,3 +85,15 @@ def _is_jwt_access_token_valid(self, access_token: str) -> bool:
...
+
+    def _is_api_key_valid(self, api_key: str) -> bool:
+        """
+        Check if the JWT access token is valid.
+
+        It does this by checking that it was signed by the corresponding private key and has not expired. It also
+        requires the payload to contain a username.
+        :param api_key: The JWT access token to check.
+        :return: `True` if the JWT access token is valid and its payload contains a username, `False` otherwise.
+        """
```
Comment (Dagonite) [inline fia_api/core/auth/tokens.py:97]: Docstring is talking about JWT instead of API key.

Comment (Pasarus) [reply fia_api/core/auth/tokens.py:97]: Good spot!

File: test/core/auth/test_tokens.py [hunk 2]
```diff
@@ -14,21 +16,31 @@
...
     assert user.user_number == expected_user_number
     assert user.role == "user"
 
 
+def test_get_user_from_token_api_key():
+    os.environ["FIA_API_API_KEY"] = "1234567"
+    user = get_user_from_token("1234567")
+    expected_user_number = -1
+    assert user.user_number == expected_user_number
+    assert user.role == "staff"
```
Comment (Dagonite) [inline test/core/auth/test_tokens.py:31]: Might be intentional but is the role meant to be staff for this test?

Comment (Pasarus) [reply test/core/auth/test_tokens.py:31]: Yes API calls should be all access

**Review by Dagonite — CHANGES_REQUESTED** [review]
Looks good just a couple of things I noticed.

**Review by Dagonite — DISMISSED** [review]

**Pull request merged**